client/blocks/privacy-policy-banner/index.jsx

```diff
@@ -89,84 +77,75 @@ class PrivacyPolicyBanner extends Component {
 		}
 
-		// check if the current policy is under the notification period.
-		const notifyFrom = moment.utc( get( privacyPolicy, 'notification_period.from' ) );
-		const notifyTo = moment.utc( get( privacyPolicy, 'notification_period.to' ) );
+		const { moment, privacyPolicy, translate } = this.props;
 
-		if (
-			( ! notifyFrom.isBefore() || ! notifyTo.isAfter() ) &&
-			! config.isEnabled( 'privacy-policy/test' )
-		) {
-			showPrivacyPolicyBanner = false;
-		}
+		if ( ! config.isEnabled( 'privacy-policy/test' ) ) {
+			// check if the user has already accepted/read the privacy policy.
+			if ( get( this.props.privacyPolicyPreferenceValue, [ privacyPolicy.id ] ) === true ) {
+				return null;
+			}
 
-		// check if the register date of the user is after the notification period
-		const userRegisterDate = moment( this.props.userRegisterDate );
+			// check if the current policy is under the notification period.
+			const notifyFrom = moment.utc( get( privacyPolicy, 'notification_period.from' ) );
+			const notifyTo = moment.utc( get( privacyPolicy, 'notification_period.to' ) );
 
-		if ( userRegisterDate.isAfter( notifyFrom ) ) {
-			return null;
+			if ( ! moment().isBetween( notifyFrom, notifyTo ) ) {
+				return null;
+			}
+
+			// check if the register date of the user is after the notification period
+			if ( moment( this.props.userRegisterDate ).isAfter( notifyFrom ) ) {
+				return null;
+			}
 		}
 
 		return (
-			<div className="privacy-policy-banner">
-				<QueryPrivacyPolicy />
-
-				{ showPrivacyPolicyBanner && (
-					<Banner
-						callToAction={ translate( 'Learn More' ) }
-						description={ this.getDescription( privacyPolicy.effective_date ) }
-						disableHref={ true }
-						icon="pages"
-						onClick={ this.openPrivacyPolicyDialog }
-						title={ translate( 'Privacy Policy Updates.' ) }
-					/>
-				) }
-
-				{ showPrivacyPolicyBanner && (
+			<Fragment>
+				<Banner
+					callToAction={ translate( 'Learn More' ) }
+					description={ this.getDescription( privacyPolicy.effective_date ) }
+					disableHref={ true }
+					icon="pages"
+					onClick={ this.openPrivacyPolicyDialog }
+					title={ translate( 'Privacy Policy Updates.' ) }
+				/>
+				{ this.state.showDialog && (
 					<PrivacyPolicyDialog
-						isVisible={ this.state.showDialog }
+						isVisible
 						content={ privacyPolicy.content }
 						title={ privacyPolicy.title }
 						version={ privacyPolicy.id }
 						onClose={ this.closePrivacyPolicyDialog }
 						onDismiss={ this.closePrivacyPolicyDialog }
 					/>
 				) }
-			</div>
+			</Fragment>
 		);
 	}
 }
 
 const mapStateToProps = state => {
-	const privacyPolicy = getPrivacyPolicyByEntity( state, AUTOMATTIC_ENTITY );
-	const privacyPolicyUserStatus = getPreference( state, PRIVACY_POLICY_PREFERENCE );
-	const privacyPolicyId = privacyPolicy.id;
+	const privacyPolicyPreferenceValue = getPreference( state, PRIVACY_POLICY_PREFERENCE );
+	const privacyPolicy = requestPrivacyPolicy().data;
```

Comment:
as this is an otherwise quite self-contained module I'd like for us to keep open the possibility of inlining the call to `requestHttpData`. That is, since it's not needed anywhere else then it seems reasonable to colocate it here with its only consumer, thus leaving fewer files and linkages in the codebase.

Should we ever need to use it in multiple places it should be easy enough to extract.

Reply:
I'm not sure I understand: do you mean removing the `state/data-getters/privacy-policy` module and moving the code together with the `PrivacyPolicyBanner` React component? Like one would co-locate a GraphQL query, for example?

Reply:
yeah that exactly. the point being that since this API call is highly specific to this and only this otherwise-self-contained React component then there's no need to split it into a separate module in a separate directory

Reply:
Totally agreed. Pushed an update.